### ISSUE TITLE
Favicon URL is not retrieved when not passed at CLI and scheme is not specified in HTML document

### ISSUE THREAD

**Issue by benoit74** [openzim/warc2zim#195]
When no favicon URL is passed as a CLI input and we try to "guess" it from html main page headers and the scheme is not specified, the resulting favicon URL is incorrect.

Not working:
```
<link rel="icon" href="//dha92jo6cen2v.cloudfront.net/styles/mefi/favicon030514b.ico" type="image/ico">
```

```
<link rel="icon" href="/styles/mefi/favicon030514b.ico" type="image/ico">
```

```
<link rel="icon" href="./styles/mefi/favicon030514b.ico" type="image/ico">
```

The cause is that the scraper tries to build the favicon URL with an `urljoin` on `self.main_url` and the value extracted from the link above, but `self.main_url` is normalized and hence does not contain the HTTP scheme anymore.

**Comment by benoit74** [comment]
In addition to this, we always download the live version even if the file is present in WARC records

**Comment by benoit74** [comment]
Fixed by #196
